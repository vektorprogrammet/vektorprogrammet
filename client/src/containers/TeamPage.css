.team-header-text{
    width:500px;
}

.gradient-box-grid{
    height:100%;
}
