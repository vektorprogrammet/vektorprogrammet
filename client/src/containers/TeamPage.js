import React, { Component } from 'react';

import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { fetchDepartments } from '../actions/department';

import TeamOverview from '../components/TeamOverview/TeamOverview';
import PageHeader from '../components/PageHeader';
import './TeamPage.css';

class TeamPage extends Component {
  componentDidMount() {
    if (!this.props.departments.length > 0) {
      this.props.fetchDepartments();
    }
  }

  render() {
    return (
      <div>
        <div className="container">
          <PageHeader>
            <h1>Team</h1>
            <p>Teamene har ansvar for alt fra rekruttering til drift av
              nettsiden, sponsorer og lignende. Alle organisasjonelle oppgaver tas h책nd om av
              frivillige teammedlemmer. <span className="medium-bold">Vi f책r vektorprogrammet til 책 g책 rundt! </span>
            </p>
          </PageHeader>
          {/*<GradientBox/>*/}
        </div>
        <div className="team-page-overview">
          <TeamOverview departments={this.props.departments}/>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  departments: state.departments,
});

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchDepartments,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(TeamPage);
