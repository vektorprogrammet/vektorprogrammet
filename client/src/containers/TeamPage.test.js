import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TeamPage from './TeamPage';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TeamPage />, div);
});

it('renders correctly', () => {
  const tree = renderer.create(<TeamPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
