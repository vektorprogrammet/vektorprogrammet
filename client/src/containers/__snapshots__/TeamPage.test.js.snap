// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<h1>
  Team page
</h1>
`;
